src/OGL3X/GLSLCombiner_ogl3x.cpp

```diff
@@ -142,18 +142,26 @@ void InitZlutTexture()
 	glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_NEAREST);
 	glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
 	glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
+
+#ifndef GLESX
 	glTexImage2D(GL_TEXTURE_2D, 0, fboFormats.lutInternalFormat,
 		512, 512, 0, fboFormats.lutFormat, fboFormats.lutType,
 		zLUT);
-	glBindImageTexture(ZlutImageUnit, g_zlut_tex, 0, GL_FALSE, 0, GL_READ_ONLY, fboFormats.lutInternalFormat);
+#else
+	glTexStorage2D(	GL_TEXTURE_2D, 1, fboFormats.lutInternalFormat, 512, 512);
+	glTexSubImage2D(GL_TEXTURE_2D, 0, 0, 0, 512, 512, fboFormats.lutFormat, fboFormats.lutType, zLUT);
```

Comment:
This must work with GL as well, so let make that code common. Please replace the old code, without ifdef endif